src/rpc/blockchain.cpp

```diff
@@ -140,6 +140,35 @@ static int ComputeNextBlockAndDepth(const CBlockIndex* tip, const CBlockIndex* b
     return blockindex == tip ? 1 : -1;
 }
 
+CBlockIndex* ParseHashOrHeight(const UniValue& param, ChainstateManager& chainman) {
+    LOCK(::cs_main);
+    CChain& active_chain = chainman.ActiveChain();
+
+    if (param.isNum()) {
+        const int height{param.get_int()};
+        if (height < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Target block height %d is negative", height));
+        }
+        const int current_tip{active_chain.Height()};
+        if (height > current_tip) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Target block height %d after current tip %d", height, current_tip));
+        }
+
+        return active_chain[height];
+    } else {
+        const uint256 hash{ParseHashV(param, "hash_or_height")};
+        CBlockIndex* pindex = chainman.m_blockman.LookupBlockIndex(hash);
+
+        if (!pindex) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+        }
+        if (active_chain.Contains(pindex)) {
```

Comment:
In 0123b7e4e2 "rpc: gettxoutsetinfo can be requested for specific blockheights" the range-diff seems to be:
```diff
-        if (!::ChainActive().Contains(pindex)) {
+        if (active_chain.Contains(pindex)) {
```
Should it be `if (!active_chain.Contains(pindex)) {` ? (it's late so I may be misreading)

Reply:
Looks like that commit and then f87cf789 are affected. Ugh, my sympathies for the rebase.

Reply:
Yepp, thanks. Fixed!